### PR TITLE
[NEW] Add tags to uploaded images using Google Cloud Vision API

### LINKED ISSUE #6019: Integration with Google Cloud Vision API
# Powerful Image Analysis

Google Cloud Vision API enables you to understand the content of an image by encapsulating powerful machine learning models in an easy to use REST API. It quickly classifies images into thousands of categories (e.g. "sailboat", "Eiffel Tower"), detects individual objects and faces within images, and finds and reads printed words contained within images.

https://cloud.google.com/vision/

### PR DESCRIPTION
**Pull request by marceloschmidt**
@RocketChat/core 

Closes #6019

image

### CODE REVIEW

**Comment by lucasvanhalst** [comment]
That's neat, will those tags be searchable?

**Comment by marceloschmidt** [comment]
Yes that's in our plans.

**Comment by marceloschmidt** [comment]
@RocketChat/core conflicts have been solved. Can I have some eyes on this?

File: packages/rocketchat-google-vision/server/googlevision.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		RocketChat.models.Settings.update({ _id: 'GoogleVision_Current_Month_Calls' }, { $inc: { value: count } });
+		return true;
+	}
+
+	blockUnsafeImages(message) {
+		if (this.enabled && this.serviceAccount && message && message.file && message.file._id) {
+			const file = RocketChat.models.Uploads.findOne({ _id: message.file._id });
+			if (file && file.type && file.type.indexOf('image') !== -1 && file.store === 'GoogleCloudStorage:Uploads' && file.googleCloudStorage) {
+				if (this.incCallCount(1)) {
+					const bucket = this.storageClient.bucket(file.googleCloudStorage.bucket);
```
Comment (sampaiodiego) [inline packages/rocketchat-google-vision/server/googlevision.js]: we don't store bucket name on the file anymore.

File: packages/rocketchat-google-vision/server/googlevision.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+					const bucketFile = bucket.file(`${ file.googleCloudStorage.path }${ file._id }`);
```
Comment (sampaiodiego) [inline packages/rocketchat-google-vision/server/googlevision.js]: the file path property changed as well

File: packages/rocketchat-google-vision/server/googlevision.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+			}
+		}
+		RocketChat.models.Settings.update({ _id: 'GoogleVision_Current_Month_Calls' }, { $inc: { value: count } });
+		return true;
+	}
+
+	blockUnsafeImages(message) {
+		if (this.enabled && this.serviceAccount && message && message.file && message.file._id) {
+			const file = RocketChat.models.Uploads.findOne({ _id: message.file._id });
+			if (file && file.type && file.type.indexOf('image') !== -1 && file.store === 'GoogleCloudStorage:Uploads' && file.googleCloudStorage) {
```
Comment (sampaiodiego) [inline packages/rocketchat-google-vision/server/googlevision.js]: we don't have the `file.googleCloudStorage` property anymore.

**Comment by rodrigok** [comment]
@marceloschmidt Can you test it again?

**Pull request merged**